src/components/toolSelfCheck/index.tsx

```diff
@@ -0,0 +1,873 @@
+                                                pointerEvents="none"
+                                                children={
+                                                    <Circle
+                                                        size="15px"
+                                                        color="white"
+                                                        borderWidth="2px"
+                                                        borderColor="black"
+                                                    ></Circle>
+                                                }
+                                            />
+                                            <Input
+                                                onChange={(e) =>
+                                                    onChangeOptionInput(
+                                                        questionId,
+                                                        index,
+                                                        e.target.value,
+                                                        "option",
+                                                    )
+                                                }
+                                                value={options[index][0]}
+                                                variant="flushed"
+                                                placeholder={`Option ${
+                                                    index + 1
+                                                }`}
+                                                mr={"6"}
+                                                isTruncated
+                                            />
+                                            <InputRightElement width="4.5rem">
+                                                <CloseButton
+                                                    name={questionId}
+                                                    onClick={() => {
+                                                        onRemoveOption(
+                                                            questionId,
+                                                            index,
+                                                        );
+                                                    }}
+                                                />
+                                            </InputRightElement>
+                                        </InputGroup>
+                                    ))}
+                                    <Button
+                                        maxWidth={"120"}
+                                        variant="ghost"
+                                        onClick={() => {
+                                            onAddOption(questionId, "bottom");
+                                        }}
+                                    >
+                                        +Add Option
```

Comment:
I think the button to add an option is repeated for every single question type- if you want and can easily do it consider moving it to a top level function so you don't need to repeat yourself